geomesa-core/src/main/scala/org/locationtech/geomesa/core/iterators/DensityIterator.scala

```diff
@@ -294,5 +219,12 @@ object DensityIterator extends Logging {
     }
     table
   }
+}
 
+case class DensityIteratorResult(geometry: Geometry,
+                                 densityGrid: SparseMatrix = HashBasedTable.create[Double, Double, Long]()) extends Result {
+  override def addToFeature(featureBuilder: SimpleFeatureBuilder): Unit = {
+    featureBuilder.add(DensityIterator.encodeSparseMatrix(densityGrid))
+    featureBuilder.add(geometry)
```

Comment:
Looking at org.locationtech.geomesa.core.index.QueryPlanner#adaptIteratorForDensityQuery and org.locationtech.geomesa.core.iterators.DensityIterator#expandFeature, it seems to be discarded except in a failure case where it is the backup geometry returned.